table/metadata.go

```diff
@@ -80,20 +92,544 @@ type Metadata interface {
 	SnapshotByName(name string) *Snapshot
 	// CurrentSnapshot returns the table's current snapshot.
 	CurrentSnapshot() *Snapshot
+	// Ref returns the snapshot ref for the main branch.
+	Ref() SnapshotRef
+	// Refs returns a map of snapshot refs by name.
+	Refs() iter.Seq2[string, SnapshotRef]
```

Comment:
We should probably update the comment since it's returning a *list* of name/snapshot ref pairs now, not a map